frontend/e-commerce/src/store/categorySlice.js

```diff
@@ -0,0 +1,62 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import {getAllCategories} from '../Api';
+import { useNavigate } from 'react-router-dom';
+
+
+
+export const loadCategories = createAsyncThunk(
+    'categories/loadCategories',
+    async () => {
+        try {
+            const results = await getAllCategories();
+            const jsonData = await results.json();
+            return jsonData;
+        } catch (e) {
+            useNavigate('/error');
+        }
+    }
+);
+
+
+const fetchData = (payload) => {
```

Comment:
why do you need this function? if I'm not missing anything it looks to me like it takes an array of objects and maps it to exactly the same array of objects.